packages/backend/src/orders/orders.gateway.ts

```diff
@@ -167,13 +169,43 @@ export class OrdersGateway implements OnGatewayInit, OnGatewayConnection, OnGate
         const clientRoom = `ordersRoomTable:${args.order.tableId}Restaurant:${args.order.restaurantId}`;
         const clients = this.users.get(clientRoom);
 
-        clients.map((c) => {
-            c.order.map((o) => {
-                if (o.id === args.order.id) {
-                    this.wss.to(c.socket).emit("orderUpdated", args.order);
-                    return;
-                }
+        if (args.order.status === "ready") {
+            const notificationMessage = `La commande N°${args.order.id} est prête !`;
+            this.sendNotificationToWaiters(args.order.restaurantId, args.order.tableId, notificationMessage);
+        }
+
+        if (clients !== undefined) {
+            clients.map((c) => {
+                c.order.map((o) => {
+                    if (o.id === args.order.id) {
+                        this.wss.to(c.socket).emit("orderUpdated", args.order);
+                        return;
+                    }
+                })
+            });
+        }
+    }
+
+    @SubscribeMessage('needSomething')
+    needSomething(client: Socket, args: Record<string, any>){
+        this.sendNotificationToWaiters(args.idRestaurant, args.idTable, args.thing);
+    }
+
+    private sendNotificationToWaiters(idRestaurant: number, idTable: number, message: string) {
+        // TODO : Retrouver l'intitulé de la table
+        this.usersService.findMultipleBy({
+            role: 'R_SERVER',
+            restaurant: idRestaurant,
+            where: {
+                expoToken: Not(IsNull())
+            }
+        }).then((users) => {
+            const expoTokens = [];
+            users.forEach((user) => {
+                expoTokens.push(user.expoToken);
             })
+            const notificationMessage = `Table ${idTable} : ${message}`;
+            this.ordersService.sendNotification(expoTokens, notificationMessage); // Également envoyer l'ID de la table ainsi que le "thing".
         });
     }
```

Comment:
pour répondre à ton TO-DO tu peux faire ceci : 
```suggestion
    private async sendNotificationToWaiters(idRestaurant: number, idTable: number, message: string) {
        // TODO : Retrouver l'intitulé de la table
        const waiters = await this.usersService.findMultipleBy({
            role: 'R_SERVER',
            restaurant: idRestaurant,
            where: {
                expoToken: Not(IsNull())
            }
        });
        const table = await this.tableService.findOne(idTable);
        waiters.forEach((waiter) => [...waiter.expoToken]));
        const notificationMessage = `${table.libelle : ${message}`;
        this.ordersService.sendNotification(expoTokens, notificationMessage); // Également envoyer l'ID de la table ainsi que le "thing".
        });
    }
```

Reply:
c'est dans l'idée après on peut garder ce qu'il y a là mais tu l'auras pour le moment où on voudra le corriger